api/app/signals/apps/services/domain/auto_create_children.py

```diff
@@ -0,0 +1,130 @@
+        response_json = response.json()
+        if response.status_code == 200 and len(response_json['_embedded']['container']) == 1:
+            coordinates = response_json['_embedded']['container'][0]['geometrie']['coordinates']
+            geometry = Point(coordinates[0], coordinates[1], srid=28992)
+            geometry.transform(ct=4326)
+            return geometry
+        return default
+
+    @staticmethod
+    def _translate_type_2_category(container_type):
+        slug = AutoCreateChildrenService._type_2_category_slug['default']
+        if container_type.lower() in AutoCreateChildrenService._type_2_category_slug:
+            slug = AutoCreateChildrenService._type_2_category_slug[container_type.lower()]
+        return Category.objects.get(slug=slug)
+
+    @staticmethod
+    def _container_data(signal):
+        answers = []
+        if signal.extra_properties:
+            for extra_property in signal.extra_properties:
+                if extra_property.get('id', '').lower() == 'extra_container':
+                    for answer in extra_property['answer']:
+                        answers.append(answer)
+        return answers
+
+    @staticmethod
+    def _is_applicable(signal):
+        if signal.is_parent or signal.is_child or signal.status.state != GEMELD:
+            return False
+
+        category_slug = signal.category_assignment.category.slug
+        if category_slug not in set(AutoCreateChildrenService._type_2_category_slug.values()):
+            return False
+
+        if len(AutoCreateChildrenService._container_data(signal)) < 2:
+            return False
+
+        return True
+
+    @staticmethod
+    def _create_child(signal, container_data):
+        category = AutoCreateChildrenService._translate_type_2_category(container_data['type'])
+        geometry = AutoCreateChildrenService._get_container_location(container_data['id'], signal.location.geometrie)
+
+        signal_data = {
+            'text': signal.text,
+            'text_extra': signal.text_extra,
+            'incident_date_start': signal.incident_date_start,
+            'parent': signal,
+            'extra_properties': [
+                {
+                    'id': 'extra_container',
+                    'label': 'Container(s)',
+                    'answer': [
+                        container_data
+                    ],
+                    'category_url': category.get_absolute_url()
+                }
+            ]
+        }
+        location_data = {
+            'geometrie': geometry,
+        }
+        status_data = {
+            'state': GEMELD,
+        }
+        category_data = {
+            'category': category,
+        }
+        reporter_data = {
+            'email': signal.reporter.email,
+            'phone': signal.reporter.phone,
+        }
+
+        # Create the child signal using the actions manager
+        Signal.actions.create_initial(signal_data, location_data, status_data, category_data, reporter_data)
```

Comment:
Would be nice to silence signals here, if possible.

Reply:
Closing this comment, it is in fact not a good idea to silence the signals (are used to update search index as well).